### ISSUE TITLE
ERROR: could not get subscription token

### ISSUE THREAD

**Issue by ScrewDrivert** [SoMuchForSubtlety/f1viewer#231]
Please check the [FAQ](https://github.com/SoMuchForSubtlety/f1viewer#Faq) and search [existing issues](https://github.com/SoMuchForSubtlety/f1viewer/issues?q=is%3Aissue) before you submit a new one!

**Describe the bug**

**To Reproduce**
1. Go to...
2. Try to play...
3. See error

**Expected behaviour**

**Screenshots**

**F1TV account plan**
Do you have a free, access or pro account? Are you using a VPN / proxy?

**Desktop (please complete the following information):**
 - OS
 - installation method
 - Version [`f1viewer -v` output]

**Logs**
If applicable please provide the relevant portion of your logs. You can find them by running `f1viewer -logs`.

**Comment by hrafaelveloso** [comment]
I can confirm this issue.
Everytime I launch f1viewer, the credentials are requested, but can't play anything because there's a error:

`ERROR: could not get video URL: Missing parameter Ascendon Token or Entitlement Token`

**Comment by ThanhDangNgo** [comment]
I have the same issue.

**Comment by PostMalone** [comment]
Same issue here. Started to happen hours before FP1. Hopefully there is a fix to this problem.

**Comment by SoMuchForSubtlety** [comment]
Looks like they remove the user agent exception for automatic login prevention. See #111 for the original discussion.

Also affects RaceControl https://github.com/robvdpol/RaceControl/issues/414

I will take some time to investigate possible solutions later today.

**Comment by SoMuchForSubtlety** [comment]
I will probably add a way to manually set the subscription token as a temporary measure. That way, you will be able to log in with your browser and retrieve the token from there.

**Comment by SoMuchForSubtlety** [comment]
I added the option to set your subscription token directly with 69ce95045ea2f09e101f817dfbb4f8705a01bacd and 1435e188fa033328063242b96cb7335fc32e099a 

Follow [these steps](https://github.com/SoMuchForSubtlety/f1viewer/wiki/Getting-your-subscription-token) to get it

**Comment by Arnvidr** [comment]
That worked just fine. Can get through the weekend with that. Any idea how long this token will be valid?

**Comment by SoMuchForSubtlety** [comment]
> Any idea how long this token will be valid?

It will expire after four days (96 hours).

**Comment by maturd** [comment]
Hi, is there a way to use this preview version on macOS? Do you plan to push an update via brew before Australian FP1 starts? Thanks

**Comment by SoMuchForSubtlety** [comment]
> Hi, is there a way to use this preview version on macOS?

Yes, download `darwin-amd64`  (x86) or `darwin-arm64` (arm), and execute it in your terminal
 
> Do you plan to push an update via brew before Australian FP1 starts?

Yes, later today

**Comment by jejupods** [comment]
Confirming this is working on MacOS Monterey 12.3, M1 Pro Macbook Pro using the darwin-arm64 preview build. As an FYI upon clicking test there was a ~10 second hang where the app was unresponsive, but eventually the credentials were accepted. 
Thanks for the quick workaround, such a great app.

**Comment by maturd** [comment]
> Confirming this is working on MacOS Monterey 12.3, M1 Pro Macbook Pro using the darwin-arm64 preview build. As an FYI upon clicking test there was a ~10 second hang where the app interface was unresponsive, but eventually the credentials were accepted.
> Thanks for the quick workaround, such a great app.

Which command did you use to launch it? I downloaded the right preview build but I didn't understand if I need to clone it or what command to use to open it. I tried "launch", "go", "run" but I suppose there aren't correct. Sorry for the (I suppose) silly question but until now I just used brew and that is completely automated 😔

**Comment by BoleMNE** [comment]
Will there be some better solution down the line?

**Comment by SoMuchForSubtlety** [comment]
New release is out https://github.com/SoMuchForSubtlety/f1viewer/releases/tag/v2.7.0

>Will there be some better solution down the line?

Hopefully, I will leave this issue open until we find a proper fix. Other projects are working with embedded browser views, but we can't do that from a terminal application. I will investigate further and post updates if I find anything.

**Comment by maturd** [comment]
> New release is out https://github.com/SoMuchForSubtlety/f1viewer/releases/tag/v2.7.0
> 
> > Will there be some better solution down the line?
> 
> Hopefully, I will leave this issue open until we find a proper fix. Other projects are working with embedded browser views, but we can't do that from a terminal application. I will investigate further and post updates if I find anything.

All is working now on M1 Max MB Pro on Monterey 12.3.1! Thank you so much. Just FWI: use Firefox on macOS to get the correct response of the by-password request. Used Safari in the beginning and the response was not the token itself

**Comment by jejupods** [comment]
> > Confirming this is working on MacOS Monterey 12.3, M1 Pro Macbook Pro using the darwin-arm64 preview build. As an FYI upon clicking test there was a ~10 second hang where the app interface was unresponsive, but eventually the credentials were accepted.
> > Thanks for the quick workaround, such a great app.
> 
> Which command did you use to launch it. I downloaded the right preview build but I didn't understand if I need to clone it or what command to use to open it. I tried "launch", "go", "run" but I suppose there aren't correct. Sorry for (I suppose) silly question but until now I just used brew and that is completely automated 😔

It looks like the Brew release has been pushed so you won't need to follow these steps, however I'll leave the response here just in case it is helpful for you or anyone else reading this in the future. 

Download the file that matches your Mac processor architecture - if you've got an M1 Mac then it would be the _darwin-arm64_ file, double click to extract it. 

Open terminal and navigate to the location of the extracted file and type:
`sudo chmod +x f1viewer-darwin-arm64` 
(the filename will be different depending on what version you download)
Enter your mac password if prompted. This will make the f1viewer-darwin-arm64 file executable.

Now you can just double click the f1viewer-darwin-arm64 file (or whatever you called it) in Finder and it will launch.

**Comment by pizza461** [comment]
> New release is out https://github.com/SoMuchForSubtlety/f1viewer/releases/tag/v2.7.0
> 
> > Will there be some better solution down the line?
> 
> Hopefully, I will leave this issue open until we find a proper fix. Other projects are working with embedded browser views, but we can't do that from a terminal application. I will investigate further and post updates if I find anything.

Thank you for the quick fix, however I cannot paste my token with v.2.7.0 on Windows 10 21H1, I cannot paste anything into any of the fields for that matter. CTRL + V does not work for password and username either and right click does nothing ofcourse. 

I have followed all the steps with Firefox, but I am stuck now not being able to paste my token :(

Am I alone?

Edit:

Oh, I have to select the token field first, then right click the tab and then paste through the menu. Works fine now, thank you! 

I usually use Race Control, so not really used to command line programs. Maybe you could add it to your instructions as I'm sure I won't be the only one with this problem.

**Comment by yagyamalik** [comment]
Am I the only one who is getting an empty response after logging in to F1TV? The response field in by-password seems to be empty and I get the error "Failed to load response data: No resource with given identifier found"

**Comment by SoMuchForSubtlety** [comment]
> Am I the only one who is getting an empty response after logging in to F1TV? The response field in by-password seems to be empty and I get the error "Failed to load response data: No resource with given identifier found"

That's a [bug](https://bugs.chromium.org/p/chromium/issues/detail?id=141129) in Chrome, you need to use Firefox

**Comment by ThanhDangNgo** [comment]
> > New release is out https://github.com/SoMuchForSubtlety/f1viewer/releases/tag/v2.7.0
> > > Will there be some better solution down the line?
> > 
> > 
> > Hopefully, I will leave this issue open until we find a proper fix. Other projects are working with embedded browser views, but we can't do that from a terminal application. I will investigate further and post updates if I find anything.
> 
> Thank you for the quick fix, however I cannot paste my token with v.2.7.0 on Windows 10 21H1, I cannot paste anything into any of the fields for that matter. CTRL + V does not work for password and username either and right click does nothing ofcourse.
> 
> I have followed all the steps with Firefox, but I am stuck now not being able to paste my token :(
> 
> Am I alone?
> 
> Edit:
> 
> Oh, I have to select the token field first, then right click the tab and then paste through the menu. Works fine now, thank you!
> 
> I usually use Race Control, so not really used to command line programs. Maybe you could add it to your instructions as I'm sure I won't be the only one with this problem.

I cannot paste token with v.2.7.0 on Windows 10 21H1, it doesn't display any menu to paste in f1viewer

1. Open f1viewer
2. Left-click token field
3. Press and hold Tab
4. Right-click token field

I try Ctrl V or hold mouse or hold Tab but nothing happen. Please help me.

**Comment by pizza461** [comment]
> I cannot paste token with v.2.7.0 on Windows 10 21H1, it doesn't display any menu to paste in f1viewer
> 
>     1. Open f1viewer
> 
>     2. Left-click token field
> 
>     3. Press and hold Tab
> 
>     4. Right-click token field
> 
> 
> I try Ctrl V or hold mouse or hold Tab but nothing happen. Please help me.

Don't hold tab, right click the tab, meaning: the tab where it says f1viewer.exe or right click left to the minimize button. There you can find a menu that says 'paste', which will then allow you to paste.

I guess I was right in saying that I wouldn't be the only one who was struggling with this xD

**Comment by pizza461** [comment]
Christ! I'm sorry to say that it has started opening .mpd files again, with no picture. So I guess this is the end for us :( We must use the web browser to watch the race :(


Appears to be a problem with DRM and I fear there won't be a fix :'(

https://github.com/robvdpol/RaceControl/issues/413

**Comment by PostMalone** [comment]
Is F1Viewer opening blank .mpd files for anyone else? 

Replays works fine, live stream sessions not.

**Comment by campodoro74** [comment]
> Appears to be a problem with DRM and I fear there won't be a fix :'(

Same here - live video is encrypted.

**Comment by telelisast** [comment]
> > Appears to be a problem with DRM and I fear there won't be a fix :'(
> 
> Same here - live video is encrypted.

Replays from fp2 as well.

**Comment by JustJoostNL** [comment]
When is there a fix for the drm?

**Comment by vidje** [comment]
I can't get the subscription token at all, the Network tab in firefox shows nothing with "api.formula1.com" as filter. Actually, the login page returns absolutely nothing.

**Comment by FxRe** [comment]
> I can't get the subscription token at all, the Network tab in firefox shows nothing with "api.formula1.com" as filter. Actually, the login page returns absolutely nothing.  

Yeah, same here. The response to that POST doesn't load, Chrome says "Failed to load response data: No resource with given identifier found".

**Comment by fherenius** [comment]
@FxRe You can copy the request that was sent and then use a program like Postman to inspect the response and get the Subscription token.
